### PR TITLE
Set Task tab as default after login

### PR DESCRIPTION
**Pull request by RafaelCastilhoCastro**
# Fixes PROJECTS/TASKS/WBS COMPONENT (BUG #14 PRIORITY LOW)

This PR intends to set the Tasks tab as default when acessing the Dashboard. So the Task tab should automatically open after the login and when returning to Dashboard after navigating the App. This is suposed to happen to all account roles except a volunteer account with **no** task wbs assign to it. Volunteer accounts **with wbs tasks** should have Tasks tab as default too.

## Related PR's
This frontend PR is related to the old [#535](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/535) frontend PR. I took over to completed it but there were too many conflicts to be safely merged so raising a new PR was safer.

## How to test:
Keep backend on development
check into this PR on frontent
run `git pull` and `npm install`
log as different account roles and test if with all of them (except volunteer with no WBS) the Tasks tab is shown first.
Assign a **WBS task** to a volunteer account and test if the Task tab is being displayed first.

image

## Note:
Please observe that it's necessary to have a **WBS task** assigned to it not just a Project. If you're not sure how to do it, please check video below.

https://user-images.githubusercontent.com/65736872/221026449-4a37f61e-a1e8-4099-9cc8-e963ceb947f7.mp4

### CODE REVIEW

**Comment by agujjar2206** [comment]
Hello @RafaelCastilhoCastro, I have tested the PR and it works as expected.

1. Admin account displays "Tasks" tab as default:


2. Volunteer account without WBS task does not display "Tasks" tab as default:


3. Volunteer account with WBS task displays "Tasks" tab as default:


Thanks.

**Review by agujjar2206 — DISMISSED** [review]

**Review by vanelnw — DISMISSED** [review]
Hi @RafaelCastilhoCastro 
Great Job, i tested this PR and its works as expected
-Frist i login with a volunteer without task assign  then the Task tab wasn't set

https://user-images.githubusercontent.com/38879488/221090489-74855a24-f64d-453d-9386-673feb4c1702.mp4

- Next i assign tasks to that volunteer user and login again then the task tab was set be default

https://user-images.githubusercontent.com/38879488/221090701-44e3ce36-c4c9-4954-a8c0-4e59d14ad40b.mp4

- I also login with another account and it's work well

**Comment by angelinatr15** [comment]
Hi @RafaelCastilhoCastro, 

I tested this PR and it displayed the task bar as default for every login except the volunteer-no-task as expected. I assigned each account with a WBS task except the volunteer-no-task account. I logged in to each account and clicked around to different tabs then returned to the dashboard to see if it would still render the task tab as default. This was the case for every account except the volunteer-no-task which was also to be expected. I started logging into the accounts in the following order: admin, role-master, mentor, manager, testing, volunteer, volunteer-no-task, and core-team. 

https://user-images.githubusercontent.com/97488347/221109285-f0379f6b-f330-45c3-8a73-8035b5c70bf1.mov

**Review by angelinatr15 — DISMISSED** [review]

**Comment by AnishPandita** [comment]
Hi, I tested the PR and its working as intended. When the user is admin, the task tab is default. When user is volunteer but has a WBS task, task tab is default. When user is volunteer but has no WBS task, current week timelog is default. 
Screenshot 2023-02-24 at 12 06 41 AM
Screenshot 2023-02-24 at 12 07 29 AM
Screenshot 2023-02-24 at 12 08 58 AM

**Review by kamtoeddy — DISMISSED** [review]
Hi @RafaelCastilhoCastro,

Tested and everything works plus the code is ok
Screenshot (515)
Screenshot (516)
Screenshot (517)
Screenshot (518)

**Review by suigunfjc — COMMENTED** [review]
Hi @RafaelCastilhoCastro , I tested the PR by logging in with three different roles as Admin, Manager and Volunteer. The task tab was displayed as expected for Admin and Manager. For the volunteer, it showed as default after I assigned it a task. Well done! 
Screenshot 2023-02-24 133636
Screenshot 2023-02-24 133827
Screenshot 2023-02-24 133933
Though there is one thing I am concerned about, in your PR there is also an update about `deleteSelectedTask` and marking tasks as done. I think there will be a conflict with #684.

**Comment by one-community** [comment]
Excellent job everyone, thank you. Rafael, resolve the merge conflicts and I'll move this one to final review.

**Comment by NyahJ** [comment]
Hi @RafaelCastilhoCastro , 

Good work! I followed your video to test for different 2 accounts, they all worked as intended.

截屏2023-02-24 下午1 48 03
截屏2023-02-24 下午1 49 30
截屏2023-02-24 下午5 20 42

**Review by NyahJ — DISMISSED** [review]
Already left comments and test result, so I am just gonna approve this.

**Comment by crystalsong2021** [comment]
I tested out the functionality and it works as expected like you mention in the PR. Nice work @RafaelCastilhoCastro 
https://user-images.githubusercontent.com/18197013/221329193-87bd3e59-0d0f-4be9-99a6-2d3a042ec94c.mov

**Comment by anant2047** [comment]
I tested the PR. It works as expected. Sharing the homepage when you login for admin, volunteer without WBS, and then assigning WBS to volunteer and opening again.
Screen Shot 2023-02-24 at 5 42 23 PM
Screen Shot 2023-02-24 at 6 06 40 PM
Screen Shot 2023-02-24 at 6 23 59 PM

**Review by anant2047 — DISMISSED** [review]
I gave minor comments. Please fix them and functionally wise it's good. Bravo!

**Comment by tingedudu8** [comment]
Hi @RafaelCastilhoCastro, I noticed that if I assigned the task to a volunteer then deleted it, the default page should not be the task tab right? However, the task tab still open automatically after returning to the dashboard.

https://user-images.githubusercontent.com/70128933/221331936-c09b88d9-12c9-4c19-895c-cdc34383ea11.mov

**Comment by vitorsaa2k** [comment]
Great PR @RafaelCastilhoCastro! I tested using the following steps:

1. WBS assigned to a user doesn't make the task tab as default
2. Assining a task to the same user and opening the dashboard again makes the task tab as default
3. Deleting the task assigned to the user doesn't make the task tab as default
4. In an admin account the task tab will always be the default tab

https://user-images.githubusercontent.com/17365161/221352558-3703c3fc-6e9c-4d0f-ac4e-46083cfad20c.mp4

**Review by vitorsaa2k — DISMISSED** [review]

File: src/components/TeamMemberTasks/TeamMemberTasks.jsx [hunk 1]
```diff
@@ -2,130 +2,183 @@
...
+                </h3>
+              </td>
+            )
+          } else {
+            return <td></td>
+          }
+        }
+
+        const markAsDone = async task => {
+          console.log('task before', task.task)
```
Comment (rauleffting) [inline src/components/TeamMemberTasks/TeamMemberTasks.jsx:156]: To delete.

**Review by rauleffting — CHANGES_REQUESTED** [review]
Hi @RafaelCastilhoCastro. Your PR was tested and everything is working quite well, as you can see below:


https://user-images.githubusercontent.com/29555732/221364276-0e5df195-78ea-49f0-ad10-1de635097a60.mp4


However, there is one debug output line in your code that must be deleted. Can you do this? Thereafter, I'm going to approve it!

Good job! Thank you very much!

File: src/components/TeamMemberTasks/TeamMemberTasks.jsx (hunk 1 above)
Comment (RafaelCastilhoCastro) [reply src/components/TeamMemberTasks/TeamMemberTasks.jsx]: Removed, thanks.

**Comment by Jorge-Nerd** [comment]
Hi @RafaelCastilhoCastro seems to me to be working correctly
image
and appears the new project in the volunteer project
image
In the branch development does not appears  task as a default
image

**Comment by mpavithr** [comment]
Hi, everything seems right and it seems to be working correctly from my end.
Dev admin has task as default
raphael_pr1
Volunteer with no tasks does not have task as default
raphael_pr2
Volunteer with tasks have it as default
raphael_pr3

**Review by rauleffting — APPROVED** [review]
Tested and approved!

**Pull request merged**